src/components/inputs/Radio.tsx

```diff
@@ -99,78 +169,106 @@ const Radio = React.forwardRef<HTMLDivElement, RadioProps>(function RadioFn(
 		padding = { bottom: 'small' },
 		size = 'medium',
 		iconColor = 'gray0',
-		value,
+		inputRef = null,
 		...rest
-	},
-	ref
+	}: RadioProps<T>,
+	ref: React.ForwardedRef<HTMLDivElement>
 ) {
-	const innerRef = useRef<HTMLDivElement>(null);
-	const radioRef = useCombinedRefs<HTMLDivElement>(ref, innerRef);
-	const [isActive, setIsActive] = useState(defaultChecked || checked || false);
+	const containerRef = useCombinedRefs<HTMLDivElement>(ref);
+	const radioInputRef = useCombinedRefs<HTMLInputElement>(inputRef);
+	const labelRef = useRef<HTMLDivElement>(null);
+	const [isChecked, setIsChecked] = useState(checked ?? defaultChecked ?? false);
+	const [id] = useState((): string => {
```

Comment:
I am not sure to understand why this is necessary and why do you need to increment RadioComponentAlias._id value

Reply:
The id is required to link the label to the input element, so that the click on the label automatically click the input. Also the label should be always linked to an input through the for attribute (htmlFor here in jsx), for accessibility reasons.
With this logic the Radio creates a uniq id, by incrementing the "static" _id field each time a component is rendered.
I used an alias just to cast the component type, because the _id field is declared for the Radio but not for RadioComponent

Reply:
It makes sense. But isn't this achievable also with a useMemo instead? It looks weird to me to use a useState when the value is never setted

Reply:
Absolutely, I'm updating it